packages/thirdweb/src/extensions/prebuilts/deploy-modular-core.ts

```diff
@@ -0,0 +1,134 @@
+import type {
+  Abi,
+} from "abitype";
+import type { ThirdwebClient } from "../../client/client.js";
+import { getContract } from "../../contract/contract.js";
+import { deployViaAutoFactory } from "../../contract/deployment/deploy-via-autofactory.js";
+import {
+  getOrDeployInfraForPublishedContract,
+} from "../../contract/deployment/utils/bootstrap.js";
+import { fetchPublishedContractMetadata } from "../../contract/deployment/publisher.js";
+import type { Chain } from "../../chains/types.js";
+import type { Account } from "../../wallets/interfaces/wallet.js";
+import { prepareContractCall } from "../../transaction/prepare-contract-call.js";
+import { resolveMethod } from "../../transaction/resolve-method.js";
+
+export type DeployModularCoreContractOptions = {
+  client: ThirdwebClient;
+  chain: Chain;
+  account: Account;
+  contractId: string;
+  contractParams: any[];
+  publisher?: string;
+  version?: string;
+  implementationConstructorParams?: unknown[];
+};
+
+/**
+ * TODO not quite ready for public consumption yet
+ * @internal
+ */
+export async function deployModularCoreContract(
```

Comment:
is this core for ANY type of contract? erc721/erc1155/erc20/etc?

Reply:
Right now testing with a mock core contract. But this function should allow deployment of any core contract.